app/main.py

```diff
@@ -1,3 +1,87 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError(f"unsupported operand type(s) for +: "
+                            f"'Distance' and {type(other)}")
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError(f"unsupported operand type(s) for +=: "
+                            f"'Distance' and {type(other)}")
+        return self
+
+    def __mul__(self, other: float) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError(f"unsupported operand type(s) for *: "
+                            f"'Distance' and {type(other)}")
+
+    def __truediv__(self, other: float) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(round(self.km / other, 2))
+        else:
+            raise TypeError(f"unsupported operand type(s) for /: "
+                            f"'Distance' and {type(other)}")
```

Comment:
`other` can also be `int` here. Add it in type hint.

`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km. Don't check for instance type here